.github/scripts/utils/async.sh

```
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+# This file contains some utility functions for executing multiple background commands
+# and waiting for them to finish, while keeping output sequential
+
+if [[ -z "${CI}" && -z "$(command -v flock)" ]]; then
+  if [[ "$OSTYPE" != 'darwin'* || -z "$(command -v brew)" ]]; then
+    echo 'This script requires flock to be installed. Please install it and try again'
+    exit 1
+  fi
+  brew install flock
```

Sourcing scripts should neither have side effects nor install utilities on the user's machine. It's better to check for the command and raise an error if it's not present so the user can install it themselves.